src/components/UserDetails/UserDetails.jsx

```diff
@@ -0,0 +1,241 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes, { instanceOf } from 'prop-types';
+import { useForm } from 'react-hook-form';
+import * as yup from 'yup';
+import { yupResolver } from '@hookform/resolvers/yup';
+import {
+  Flex,
+  FormControl,
+  FormLabel,
+  Input,
+  Button,
+  Heading,
+  Stack,
+  Box,
+  HStack,
+} from '@chakra-ui/react';
+
+import styles from './UserDetails.module.css';
+import { passwordRequirementsRegex } from '../../utils/utils';
+import {
+  getUserFromDB,
+  updateUser,
+  logout,
+  useNavigate,
+  getCurrentUserId,
+} from '../../utils/AuthUtils';
+import { Cookies, withCookies, cookieKeys } from '../../utils/CookieUtils';
+import AUTH_ROLES from '../../utils/AuthConfig';
+
+const UserDetails = ({ userId, cookies }) => {
+  const [user, setUser] = useState({
+    firstName: '',
+    lastName: '',
+    email: '',
+    phoneNumber: '',
+  });
+  const [canEditForm, setCanEditForm] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const navigate = useNavigate();
+
+  const { SUPERADMIN_ROLE, ADMIN_ROLE, DRIVER_ROLE } = AUTH_ROLES.AUTH_ROLES;
+
+  useEffect(() => {
+    const fetchUserFromDB = async () => {
+      const userFromDB = await getUserFromDB(userId);
+
+      const currentUserRole = cookies.get(cookieKeys.ROLE);
+      const currentUserId = getCurrentUserId();
+
+      if (currentUserId !== userId) {
+        if (
+          currentUserRole === DRIVER_ROLE ||
+          (currentUserRole === ADMIN_ROLE &&
+            [SUPERADMIN_ROLE, ADMIN_ROLE].includes(userFromDB.role))
+        ) {
+          try {
+            await logout('/login', navigate, cookies);
+          } catch (err) {
+            setErrorMessage(err.message);
+          }
+        }
+      }
+      setUser(userFromDB);
+    };
+    fetchUserFromDB();
+  }, []);
+
+  const formSchema = yup.object().shape({
+    firstName: yup.string().required('Please enter your first name'),
+    lastName: yup.string().required('Please enter your last name'),
+    phoneNumber: yup
+      .string()
+      .length(10, 'Please enter a ten digit phone number')
+      .matches(/^\d{10}$/)
+      .required('Please enter your phone number'),
+    newPassword: yup
+      .string()
+      .nullable()
+      .transform(value => value || null)
```

Comment:
I don't think we need this `transform` function since we're not going to be changing the formatting of the password or anything.

Reply:
Resolved: This is needed since the validation converts the input value to `''` if the field is clicked.